### PR TITLE
Avoid jittering when player is attached

### LINKED ISSUE #10985: Movement of attached player is jerky sometimes.
##### Minetest version
```
minetest-5.4.0-7832b6843-win64
```

##### Summary
#10952 seems to have fixed the biggest problems, particularly player attachments detaching, but there's still room for improvement.
Player attachment's movement is sometimes jerky, probably based on the sequence of objects on the object list.

##### Steps to reproduce
1. create an entity with this callback:
```
on_rightclick = function(self,clicker)
	local pos = self.object:get_pos()
	local at1 = minetest.add_entity(pos,self.name)
	at1:set_attach(self.object,nil,{x=0,y=0,z=-20})
	clicker: set_attach(at1,nil,{x=0,y=10,z=0})
	self.object:set_velocity({z=30.0,y=0,x=0.0})
end,
```
2. spawn your entity and immediately right click it. player should attach and they start moving smoothly.
3. spawn your entity, then leave the game, then log in again and right click the entity, now the movement is jerky.

Now we know that it is expected that an attachment's position equals the parent's position, but based on the sequence on the list, attachment's position may be one server step late, so server position updates are outdated sometimes, so the hypothesis is this is the cause of the issue.

Maybe instead of attachments setting their own position, the parent should set position on behalf of their children recursively? That should get them in sync.

### PR DESCRIPTION
**Pull request by lhofhansl**
When a player is attached to an entity there sometimes is a lot of jittering - as if player and entity are out of sync.
(Note I am not referring to the jerkiness w.r.t. to the world, but w.r.t. to the 

A possible explanation is that entities are sorted differently on the the client and the server.

*Potentially* fixes #10985

## To do

Testing! You can try any mod that attach a player (hot air balloon, steam punk blimp, etc).
Notice that *sometimes* movement is very jittery. Not w.r.t. to the world but w.r.t. to the attached entity.

(I think different sort orders on client and server might be the problem, but I have not finally convinced myself that this is the case.
By changing the sort order I might also be just lucky in all cases for accidentally have one that does not cause jittering)

## How to test

See above.

### CODE REVIEW

**Comment by appgurueu** [comment]
Is `updateAbsoluteTransform()` called on the parents before being drawing the children? Otherwise you might have a child relative to a transform that is one frame old.

**Comment by lhofhansl** [comment]
@appgurueu I'd need to check. With this change I have not observed any jitter.

**Comment by appgurueu** [comment]
@lhofhansl reason for the close? Did you figure out this was insufficient?

**Comment by lhofhansl** [comment]
Before I closed the PR I had tested to reverse the iteration order and then see whether I would now always see the rubberband effect. And I did not. That's when I closed this, because it seemed like I did not understand what was going on.

**Comment by lhofhansl** [comment]
Reopening. I had this one for a months now, and have not observed any severe stutter (which was common before)

**Comment by x2048** [comment]
I can confirm this fixes #10985 when restarting the game while flying in hydroplane.

**Review by numberZero — COMMENTED** [review]
The change is trivial and unlikely to have any adverse side effects.

File: src/activeobjectmgr.h [hunk 1]
```diff
@@ -38,7 +38,7 @@ class ActiveObjectMgr
 
 	T *getActiveObject(u16 id)
 	{
-		typename std::unordered_map<u16, T *>::const_iterator n =
+		typename std::map<u16, T *>::const_iterator n =
 				m_active_objects.find(id);
```
[the remark this answers is not in the document]
Comment (lhofhansl) [reply src/activeobjectmgr.h:42]: +1

**Review by sfan5 — APPROVED** [review]

**Review by SmallJoker — APPROVED** [review]
Looks about right.

**Pull request merged**